src/core/store.js

```diff
@@ -4,6 +4,16 @@ import { applyMiddleware, compose } from 'redux';
 import createLogger from 'redux-logger';
 import config from 'config';
 
+
+// HEY TOFUMATT THIS IS WHAT YOU WANT
+const tofuWare = store => next => action => {
```

Comment:
Does the [`onEnter`](https://github.com/ReactTraining/react-router/blob/master/docs/Glossary.md#enterhook) hook from react-router always get called? I think you might be able to add one to the `App` route instead of switching to react-redux-router.